docs/developer/decision-records/2022-06-20-mvd-onboarding/README.md

```diff
@@ -0,0 +1,36 @@
+# Participants onboarding in MVD
+
+This document outlines the infrastructure steps for the scenario of onboarding participants to the Minimum Viable Dataspace (MVD).
+
+## Precondition - GAIA-X membership
+
+To join the MVD, each participant must have a Verifiable Credential (VC) signed by GAIA-X Authority which proves GAIA-X membership.
+
+At present the MVD deployment pipeline does not interact with the real GAIA-X Authority. A simulated GAIA-X Authority is deployed within MVD for demonstration purposes.
+
+## Onboarding process - GAIA-X membership verification
+
+During the onboarding process, the GAIA-X membership claim is verified by the Dataspace Authority.
+The onboarding flow is presented in the [distributed authorisation sub-flow document](https://github.com/agera-edc/MinimumViableDataspace/tree/feature/20-rs-adr-target/docs/developer/decision-records/2022-06-16-distributed-authorization).
+Actors in the onboarding scenario:
+
+- _Participant A_ is a putative participant that wants to join the MVD.
+- _Participant B_ is the Dataspace Authority (Registration Service and its `did:web` document).
+- _Authority_ is the simulated GAIA-X Authority.
+
+## Infrastructure
+
+To enable above precondition and onboarding scenario in MVD, the following steps need to be implemented in the MVD infrastructure: 
+
+1. The MVD deployment workflow will generate the private and public keys for GAIA-X Authority.
+2. The MVD deployment workflow will deploy the GAIA-X Authority DID document containing the public key.
+3. The GAIA-X private key will be used in the participant deployment workflow to generate GAIA-X membership Verifiable Credentials. Additionally, 
+   it will be published as Github artifact so that it can be used locally to onboard additional participants.
+4. A CLI client for IdentityHub will be implemented. It will be used in the MVD deployment workflow to populate the participant's Identity Hub with GAIA-X 
+   membership Verifiable Credentials.
+5. A CLI client for the Registration Service will be implemented. It will be used in the MVD deployment workflow to start the participant onboarding process. The CLI
+   client can also be used locally to onboard additional participants. The CLI client needs access to the participant's DID private key.
+6. The Registration Service will be configured at deployment with the environment variable pointing to GAIA-X Authority DID URL. This is required because each deployment will have a different URL for the GAIA-X Authority.
```

Comment:
Just a question: 

Didn't fully understood why "each deployment will have a different URL for the GAIA-X Authority." ? What is the purpose of this one ?

Reply:
@cpeeyush The GAIA-X Authority will be implemented in the form of a DID document, which is hosted at a specific URL. Since every deployment has a different DID document (due to the keys used), every deployment needs to have it's own URL where the DID document is hosted.

Reply:
okay but I was thinking GAIA-X Authority can be deployed once during very first infra setup (where we deploy ACR and storage accounts for terraform states) and this can be reused for every other MVD dataspace.

This will be closer to future scenario when we move to start using real Gaia-X authority service as that service is always there and not redeployed with a dataspace deployment.

But as discussed with @Izzzu this is not to hold this PR, we can merge this PR as it is and can discuss this later.

(Anyways this is just an idea as we are using a mock Gaia-X authority here)

Reply:
@cpeeyush as we discussed I added an explanation in point 5 to make clear that the approach of deploying the GAIA-X Authority every time with the Dataspace is a simplification, but in general it's an external central component:

`For the simplicity the GAIA-X Authority DID document 
   is deployed together with other Dataspace components, which means that each deployed Dataspace will have its own GAIA-X Authority instance. In a
   real scenario the GAIA-X Authority is an external component and can communicate with multiple Dataspaces.`
   
   Please take a look :)

Reply:
@cpeeyush that's a great catch! I think both approaches work, while a single GAIA-X authority is closer to reality. let's decide on one approach and then move on.